resources_measurement/linux_resources/total_memory_usage.py

```
@@ -0,0 +1,74 @@
+from resources_measurement.linux_resources.total_resource_usage import LinuxContainerResourceReader
+
+# Constants for paths and identifiers
+MEMORY_USAGE_FILE_NAME_V2 = "memory.current"
+MEMORY_USAGE_FILE_NAME_V1 = "memory/memory.usage_in_bytes"
+
+MEMORY_MAX_FILE_NAME_V2 = "memory.max"
+MEMORY_MAX_FILE_NAME_V1 = "memory/memory.limit_in_bytes"
+
+HOST_PROC_MEMORY_LIMIT_FILE = r"/proc/meminfo"
+MEM_TOTAL_PREFIX = "MemTotal:"
+
+
+class LinuxContainerMemoryReader(LinuxContainerResourceReader):
+    def __init__(self):
+        super().__init__()
+        self.__memory_limit = self.__get_memory_limit_bytes(self._version)
+
+    def get_memory_usage_bytes(self) -> int:
+        try:
+            with open(self._resource_usage_path) as f:
+                return int(f.read().strip())
+        except Exception as e:
+            print(f"Error reading memory usage: {e}")
+            return 0
+
+    def get_memory_usage_percent(self) -> float:
+        usage = self.get_memory_usage_bytes()
+        if self.__memory_limit <= 0:
+            return 0.0  # TODO: check if this is the right value
+        return (usage / self.__memory_limit) * 100
+
+    def _get_resource_file_path(self, version: str) -> str:
+        cgroup_dir = extract_cgroup_relative_path(version, FileKeywords.CGROUP_V1_MEMORY_IDENTIFIER)
+        if version == FileKeywords.V2:
+            return os.path.join(cgroup_dir, MEMORY_USAGE_FILE_NAME_V2)
+        else:
+            return os.path.join(cgroup_dir, MEMORY_USAGE_FILE_NAME_V1)
+
+    def __extract_memory_limit_file(self, version: str) -> str:
+        cgroup_dir = extract_cgroup_relative_path(version, FileKeywords.CGROUP_V1_MEMORY_IDENTIFIER)
+        if version == FileKeywords.V2:
+            return os.path.join(cgroup_dir, MEMORY_MAX_FILE_NAME_V2)
+        else:
+            return os.path.join(cgroup_dir, MEMORY_MAX_FILE_NAME_V1)
+
+    def __get_host_memory_limit(self) -> int:
```

I dont understand the whole host fallback thing.
In case we cant use cgroups for our purpose - just use psutil, or even better, raise exception. Don't you think?

---

what do you mean?
Yes, we can use psutil as discussed
why should we raise an error?